### PR TITLE
Related items

### PR DESCRIPTION
**Pull request by callumreid**
-Formatted placement of related products action button and changed it to a star as atelier dictates. Still doesn't look nice, but does meet required specs
-Modified addToOutfit function so that it does not allow duplicate items to be added to outfit
- Changed related products list to only render 3 items at a time. Buttons conditionally appear to load next or previous items. Buttons scroll list by 1 product as atelier requires

Lots of commits are just preparing for merge/merges, so they don't really need to be reviewed. Main addition is related product list carousel functionality with conditional buttons, and a couple of bug fixes/ styling tweaks

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (rlawrence9) [inline .vscode/settings.json:1]: is this needed?

File: client/src/relatedProducts/RelatedProductsView/RelatedProducts.jsx [hunk 2]
```diff
@@ -22,12 +22,15 @@ export default class RelatedProducts extends React.Component {
...
       modifiedCurrent: {},
-      features: []
+      features: [],
+      displayedProductsIndices: [0, 1, 2]
     }
     this.handleAddToOutfit = this.handleAddToOutfit.bind(this);
     this.handleRemoveFromOutfit = this.handleRemoveFromOutfit.bind(this);
     this.handleCompareItems = this.handleCompareItems.bind(this);
     this.closeModal = this.closeModal.bind(this);
+    this.handlePrevClick = this.handlePrevClick.bind(this);
```
Comment (rlawrence9) [inline client/src/relatedProducts/RelatedProductsView/RelatedProducts.jsx:32]: not anything that needs to even be done, but could these two functions be combined into one handler to save space?  theoretically the clicks could be separated by a condition which decides which state to update maybe?

File: client/src/relatedProducts/RelatedProductsView/RelatedProducts.jsx [hunk 3]
```diff
@@ -196,6 +203,27 @@ export default class RelatedProducts extends React.Component {
       });
     }
 
+    handlePrevClick() {
```
Comment (rlawrence9) [inline client/src/relatedProducts/RelatedProductsView/RelatedProducts.jsx:206]: digging the cleanly spaced code 🔥

File: client/src/relatedProducts/RelatedProductsView/RelatedProductsList.jsx [hunk 4]
```diff
@@ -4,37 +4,65 @@ import RelatedProductsCard from './RelatedProductsCard.jsx';
...
+        photo={product.photoUrl.thumbnail_url}
+        id={product.itemId}
+        features={product.features}
+        handleProductChange={props.handleProductChange}
+        handleCompareItems={props.handleCompareItems}
+        starRating={product.starRating}
+       />
+     </div>
+  })
+  let prevButton;
```
Comment (rlawrence9) [inline client/src/relatedProducts/RelatedProductsView/RelatedProductsList.jsx:24]: just food for thought - to save space, you could filter the prevButton and nextButton functionality to your helpers.  You can pass the helper function needed from main component to use here... or you can pass a function from your main component to call this functionality in helper

**Review by rlawrence9 — APPROVED** [review]
reviewed, commented, and approved by Rob - looking good callum, just left some minor comments on some late night improvements you can make (if you feel like it) 🤙

File: client/src/relatedProducts/RelatedProductsView/RelatedProducts.jsx [hunk 5]
```diff
@@ -149,11 +152,15 @@ export default class RelatedProducts extends React.Component {
 
     handleAddToOutfit (outfitItem, e) {
       e.preventDefault();
-      localStorage.setItem(outfitItem.product_id, JSON.stringify(outfitItem));
-      this.setState({
-        yourOutfitItems: [...this.state.yourOutfitItems, outfitItem]
-      })
 
+      if (this.state.yourOutfitItems.some(({product_id}) => product_id === outfitItem.product_id)) {
```
[the remark this answers is not in the document]
Comment (callumreid) [reply client/src/relatedProducts/RelatedProductsView/RelatedProducts.jsx:156]: Smart move, I sure could!

File: client/src/relatedProducts/RelatedProductsView/RelatedProductsList.jsx (hunk 4 above)
Comment (callumreid) [reply client/src/relatedProducts/RelatedProductsView/RelatedProductsList.jsx:24]: Ahh that is a good plan. I feel like I always write out everything in components at first, before eventually getting annoyed enough at seeing all the code to hide it all away in helpers

**Pull request merged**